src/app/shared/services/books.service.ts

```diff
@@ -0,0 +1,61 @@
+import { Injectable } from '@angular/core';
+import { AngularFirestore } from '@angular/fire/firestore';
+import { ReadBook } from '@app/shared/models/read-book.model';
+import { MatSnackBar } from '@angular/material/snack-bar';
+import { BookMessages } from '../enums';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class BooksService {
+
+  constructor( private firestore: AngularFirestore, private snackBar: MatSnackBar ) { }
+
+  getReadBooks(){
+      return this.firestore.collection('readBooks').snapshotChanges();
+  }
+
+  addReadBook(data:ReadBook){
+      let today = new Date();
+      data.date = today.toLocaleDateString();
+      return new Promise<any>((resolve, reject) => {
```

Comment:
At first, we don't need to create promise here.
Secondly, pls wrap all promises from third-party libraries to observables using `from` operator, coz we're using Observables all over in Angular.